pkg/util/provider/machinecontroller/machine_safety.go

```diff
@@ -161,6 +168,42 @@ func (c *controller) isAPIServerUp() bool {
 	return true
 }
 
+//AnnotateNodesUnmanagedByMCM checks for nodes which are not handled by MCM and annotes them
+func (c *controller) AnnotateNodesUnmanagedByMCM() (machineutils.RetryPeriod, error) {
+	//list all the nodes on target cluster
+	nodes, err := c.nodeLister.List(labels.Everything())
+	if err != nil {
+		klog.Errorf("Safety-Net: Error getting nodes")
+		return machineutils.LongRetry, err
+	}
+	//getMachineFromNode()
+	for i := range nodes {
+		node := nodes[i]
```

Comment:
```suggestion
		node := &nodes[i]
```
Otherwise it would still be a shallow-copy.